loudml/prometheus.py

```diff
@@ -0,0 +1,392 @@
+        # http://0.0.0.0:9090/api/v1/query_range?query=avg(go_memstats_alloc_bytes{instance=%22localhost:9090%22,%20job=%22prometheus%22})&start=1584783120&end=1584786720&step=15
+        """
+        if not isinstance(queries, list):
+            queries = [queries]
+
+        results = []
+
+        for q in queries:
+            time_pred = _build_time_predicates(q["start"], q["end"])
+            aggregator = AGGREGATORS.get(q["aggregator"])
+            if aggregator:
+                query_url = "{}/api/v1/query_range?{}&step={}&query={}({}{})".format(
+                    self.url,
+                    time_pred,
+                    q["step"],
+                    aggregator,
+                    q["metric_name"],
+                    q["tags"]
+                )
+            else:
+                query_url = "{}/api/v1/query_range?{}&step={}&query={}{}".format(
+                    self.url,
+                    time_pred,
+                    q["step"],
+                    q["metric_name"],
+                    q["tags"]
+                )
+
+            try:
+                resp = self.session.get(query_url)
+                self.session.close()
+            except requests.exceptions.SSLError as exn:
+                logging.error('Prometheus SSL error', str(exn))
+
+            if resp.ok:
+                results.append(PrometheusResult(resp.json()))
+            else:
+                logging.error('Prometheus error',
+                    resp.status_code, resp.text[:200])
+
+        return results
+
+
+    def put(self, entry):
+        """
+        Store a point in database
+        """
+        raise errors.NotImplemented("Prometheus bucket: not implemented yet")
+
+
+class PrometheusBucket(Bucket):
+    """
+    Prometheus bucket
+    """
+
+    SCHEMA = Bucket.SCHEMA.extend({
+        Required('addr'): str,
+        Optional('user', default=""): str,
+        Optional('password', default=""): str,
+        Optional('use_ssl', default=False): Boolean(),
+        # Useful if the server is using self-signed certificates
+        Optional('verify_ssl', default=False): Boolean(),
+        Optional('ssl_cert_path', default=""): str,
+    })
+
+    def __init__(self, cfg):
+        cfg['type'] = 'prometheus'
+        super().__init__(cfg)
+        self._prometheus = None
+
+    @property
+    def addr(self):
+        return self.cfg['addr']
+
+    @property
+    def user(self):
+        return self.cfg['user'] or ''
+
+    @property
+    def password(self):
+        return self.cfg['password'] or ''
+
+    @property
+    def use_ssl(self):
+        return self.cfg.get('use_ssl') or False
+
+    @property
+    def verify_ssl(self):
+        return self.cfg.get('verify_ssl') or False
+
+    @property
+    def ssl_cert_path(self):
+        return self.cfg.get('ssl_cert_path') or ''
+
+    @property
+    def prometheus(self):
+        if self._prometheus is None:
```

Comment:
`if not self._prometheus:`

If you install flake8 the "is None" comparison should trigger PEP8 warnings. The above change is a quick fix.

I use flake8 a lot. It's very useful!